### DISCUSSION TITLE
Provide reference to original content node when generating sitemap nodes

### DISCUSSION THREAD

**Discussion by ianleeder** [patrickdemooij9/SeoToolkit.Umbraco#141] [Ideas]
We would like the ability to hide individual content nodes from the sitemap, not just document types.  I've added a flag to my master document type.

I'm looking at the [`GenerateSitemapNodeNotification`](https://seotoolkit.gitbook.io/useotoolkit/extensions/notifications#generatesitemapnodenotification), but the event object doesn't provide an easy way to reference the original content node (it includes a URL, but I can't see a `ContentService.GetByUrl` method), so I can't read my flag.

I've tracked the event source down to the `GetSelfAndChildren` method in `SitemapGenerator`, and each sitemap node is created from a content node (`IPublishedContent`).  Would it be feasible (particularly with respect to performance) to include the source `IPublishedContent` in the `SitemapNodeItem`?  Then in my notification event I could simply write:
```cs
notification.Node.HideFromSitemap |= notification.Node.SourceContent.GetPropertyValue<bool>("hideFromSitemap");
```
This also seems like it would be more performant than having to do a Content lookup on each node (eg by URL or id).

Or am I going about this the wrong way?

**Comment by patrickdemooij9** [comment]
I think it is indeed smarter for the system to include the current node in the notification, especially if you want to hide it from the sitemap. I'll create an issue from this discussion so that I can pick it up
